### ISSUE TITLE
Unable to debug in container

### ISSUE THREAD

**Issue by richardrflores** [microsoft/vscode-docker#2989]
I am unable to debug in a container. It appears as if VsDbg is not getting installed when I run the debugger in VS Code. I'm not sure why.

Does this occur consistently? Yes
Repro steps:

1. Follow the instructions to debug in container [here](https://code.visualstudio.com/docs/containers/quickstart-aspnet-core#_debug-in-container)
2. When debug is started I get the error below:

```
Installing VsDbg, Runtime = linux-x64, Version = latest...
chmod +x "/Users/User1/.vsdbg/getvsdbg.sh" && "/Users/User1/.vsdbg/getvsdbg.sh" -v latest -r linux-x64 -l "/Users/User1/.vsdbg/linux-x64/latest"
Info: Previous installation at '/Users/User1/.vsdbg/linux-x64/latest' not found
Info: Using vsdbg version '17.0.10413.12'
Using arguments
    Version                    : 'latest'
    Location                   : '/Users/User1/.vsdbg/linux-x64/latest'
    SkipDownloads              : 'false'
    LaunchVsDbgAfter           : 'false'
    RemoveExistingOnUpgrade    : 'false'
Info: Using Runtime ID 'linux-x64'

ERROR: No internet connection.
Error: Process 'chmod +x "/Users/User1/.vsdbg/getvsdbg.sh" && "/...' exited with code 1
Error: 

```
Action: docker-launch
Error type: 1
Error Message: Process 'chmod +x "/Users/---/.vsdbg/getvsdbg.sh" && "/...' exited with code 1
Error: 

Version: 1.13.0
OS: darwin
OS Release: 20.3.0
Product: Visual Studio Code
Product Version: 1.56.2
Language: en

<details>
<summary>Call Stack</summary>

```
ChildProcess.<anonymous> extension.bundle.js:2:4956333
ChildProcess.emit events.js:315:20
maybeClose child_process.js:1048:16
Socket.<anonymous> child_process.js:439:11
Socket.emit events.js:315:20
Pipe.<anonymous> net.js:673:12
```

</details>

**Comment by karolz-ms** [comment]
@richardrflores are you using an M1 Mac by any chance?

**Comment by richardrflores** [comment]
No @karolz-ms Intel processor.

**Comment by karolz-ms** [comment]
@richardrflores I am unable to reproduce--I can debug with no issues.

Note that for .NET debugging the debugger is installed on developer machine, and then bind-mounted into the container, together with application . From your logs it seems that it is `User1` that is running VS Code process. Is that how your account is named? Are you running in remote-container (devcontainer) configuration? Does `User1` have network access? Network access is required to install the debugger.

**Comment by richardrflores** [comment]
@karolz-ms I substituted my account name with `User1` for this post. The user is my main account for my MBP and an admin so I assume has network access. Also, I'm not using a devcontainer.

Since you cannot reproduce the problem, I suspect the issue is environmental. I'm able to reproduce with another test project on the same machine following the same steps. 

I've tried uninstalling and re-installing the extension. I disabled the Microsoft devcontainer extension. Still same issue. I'm not sure what else I can do. Any other suggestions?

**Comment by richardrflores** [comment]
Ok. Finally got this working. Took a look at the script in `/Users/User1/.vsdbg/getvsdbg.sh` and saw there is a function called called check_internet_connection that was the source of the error. See line 402. 
![image](https://user-images.githubusercontent.com/1383703/121815758-9d5b2680-cc3d-11eb-8d36-9fdbe90238f7.png)

Although I'm not entirely sure what the script is doing it looks like it's trying to make some kind of request either using wget or  curl. Long story short, I was using the macOS version of curl as the default. After installing curl from homebrew and setting it on my path, VsDbg installed correctly. Great...but then I got a different error: Failed To Resolve With FrontEnd DockerFIle.v0

To fix that issue I had to uncheck the `Use new virtualization framework` option and I was able to debug with no issue.
![image](https://user-images.githubusercontent.com/1383703/121816098-3b9bbc00-cc3f-11eb-89d7-ce897bb51d9b.png)

Here are some SO posts that help me resolve the issue.
https://stackoverflow.com/questions/36081761/how-to-update-curl-on-osx-el-capitan
https://stackoverflow.com/questions/64221861/failed-to-resolve-with-frontend-dockerfile-v0

Hope this helps anyone else encountering this issue.
